tests/test_explorer.py

```diff
@@ -0,0 +1,172 @@
+
+    def tearDown(self):
+        del self.qapp
+
+    @mock.patch("iquip.apps.explorer.ExplorerApp.lazyLoadFile")
+    def test_file_tree_item_expanded(self, mockedLazyLoadFile):
+        app = explorer.ExplorerApp(name="name", masterPath="masterPath", parent=QObject())
+        directoryItem = QTreeWidgetItem(app.explorerFrame.fileTree)
+        directoryItem.setText(0, "directory")
+        QTreeWidgetItem(directoryItem)  # Add an empty item to an unloaded directory.
+        directoryItem.setExpanded(True)
+        mockedLazyLoadFile.assert_called_once()  # pylint: disable=no-member
+
+    @mock.patch("iquip.apps.explorer.ExplorerApp.loadFileTree")
+    def test_reload_button_clicked(self, mockedLoadFileTree):
+        app = explorer.ExplorerApp(name="name", masterPath="masterPath", parent=QObject())
+        QTest.mouseClick(app.explorerFrame.reloadButton, Qt.LeftButton)
+        # Once when the app is created, once explicitly.
+        self.assertEqual(mockedLoadFileTree.call_count, 2)  # pylint: disable=no-member
+
+    @mock.patch("iquip.apps.explorer.ExplorerApp.openExperiment")
+    def test_open_button_clicked(self, mockedOpenExperiment):
+        app = explorer.ExplorerApp(name="name", masterPath="masterPath", parent=QObject())
+        QTest.mouseClick(app.explorerFrame.openButton, Qt.LeftButton)
+        mockedOpenExperiment.assert_called_once()  # pylint: disable=no-member
+
+
+class FileFinderThreadTest(unittest.TestCase):
+    """Unit tests for _FileFinderThread class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_init_thread(self):
+        widget = QTreeWidgetItem()
+        callback = mock.MagicMock()
+        parent = QObject()
+        with mock.patch("iquip.apps.explorer._FileFinderThread.fetched") as mockedFetched:
+            thread = explorer._FileFinderThread(path="path", widget=widget,
+                                                callback=callback, parent=parent)
+            self.assertEqual(thread.path, "path")
+            self.assertEqual(thread.widget, widget)
+            mockedFetched.connect.assert_called_once_with(callback, type=Qt.QueuedConnection)
+
+    @mock.patch("iquip.apps.explorer.cmdtools.run_command")
+    def test_run(self, mocked_run_command):
+        mocked_run_command.return_value.stdout = "path1\npath2\n"
+        widget = QTreeWidgetItem()
+        parent = QObject()
+        with mock.patch("iquip.apps.explorer._FileFinderThread.fetched") as mockedFetched:
+            thread = explorer._FileFinderThread(path="path", widget=widget,
+                                                callback=mock.MagicMock(), parent=parent)
+            thread.run()
+            thread.wait()
+            mocked_run_command.assert_called_once_with("artiq_client ls path")
+            mockedFetched.emit.assert_called_once_with(["path1", "path2"], widget)
+
+
+class ExplorerAppTest(unittest.TestCase):
```

Comment:
I found that `ExplorerApp` also does not have the docstring for attributes. Please add it!

Reply:
Oh, I added only the docstring about `ExplorerFrame`.. I will modify it.